tests/integration/cloud/end_to_end/conftest.py

```diff
@@ -37,31 +41,119 @@ def context() -> CloudDataContext:
+        expectation_suite=expectation_suite
+    )
+    assert len(expectation_suite.expectations) > 0
+    _ = context.get_expectation_suite(expectation_suite_name=expectation_suite_name)
+    context.delete_expectation_suite(expectation_suite_name=expectation_suite_name)
+    with pytest.raises(gx_exceptions.DataContextError):
+        _ = context.get_expectation_suite(expectation_suite_name=expectation_suite_name)
 
 
 @pytest.fixture(scope="module")
-def get_missing_data_asset_error_type() -> type[Exception]:
-    return LookupError
+def validator(
+    context: CloudDataContext,
+    batch_request: BatchRequest,
+    expectation_suite: ExpectationSuite,
+) -> Iterator[Validator]:
+    validator: Validator = context.get_validator(
+        batch_request=batch_request,
+        expectation_suite_name=expectation_suite.expectation_suite_name,
+    )
+    validator.head()
+    yield validator
+    validator.save_expectation_suite()
+    expectation_suite = validator.get_expectation_suite()
+    _ = context.get_expectation_suite(
+        expectation_suite_name=expectation_suite.expectation_suite_name,
+    )
 
 
 @pytest.fixture(scope="module")
-def get_missing_expectation_suite_error_type() -> type[Exception]:
-    return gx_exceptions.DataContextError
+def checkpoint(
+    context: CloudDataContext,
+    batch_request: BatchRequest,
+    expectation_suite: ExpectationSuite,
+) -> Iterator[Checkpoint]:
+    checkpoint_name = (
+        f"{batch_request.data_asset_name} | {expectation_suite.expectation_suite_name}"
+    )
+    _ = context.add_checkpoint(
+        name=checkpoint_name,
+        validations=[
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            },
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            },
+        ],
+    )
+    _ = context.add_or_update_checkpoint(
+        name=checkpoint_name,
+        validations=[
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            }
+        ],
+    )
+    checkpoint = context.get_checkpoint(name=checkpoint_name)
+    assert (
+        len(checkpoint.validations) == 1
+    ), "Checkpoint was not updated in the previous method call."
+    yield checkpoint
+    # PP-691: this is a bug
+    # you should only have to pass name
+    context.delete_checkpoint(
+        # name=checkpoint_name,
+        id=checkpoint.ge_cloud_id,
+    )
+    with pytest.raises(gx_exceptions.DataContextError):
+        context.get_checkpoint(name=checkpoint_name)
 
 
 @pytest.fixture(scope="module")
-def get_missing_checkpoint_error_type() -> type[Exception]:
-    return gx_exceptions.DataContextError
+def tmp_dir(tmpdir_factory) -> py.path:
+    return tmpdir_factory.mktemp("project")
```

Comment:
`tmpdir` and `tmpdir_factory` are now "legacy" and pytest recommends using `tmp_path` and `tmp_path_factory`. You can use them the same way, they just return modern `pathlib.Path` objects instead of `py.path`.

https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures